core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -280,8 +284,19 @@ public synchronized void addMutation(TableId table, Mutation m)
     mutations.addMutation(table, m);
     totalAdded++;
 
-    if (mutations.getMemoryUsed() >= maxMem / 2) {
+    if (mutations.getMemoryUsed() >= maxMem / MAX_MEMORY_DIVISOR) {
```

Comment:
This triggers processing at 1/20th of the available memory utilization. What's better about processing at 5% of user-specified max memory instead of half? This seems like it could cause the configured memory to be under-utilized (by up to 95%).

Also, it's not clear to me the value here... since the user could simply lower the maxMemory in their BatchWriterConfig to achieve a similar effect, couldn't they?

Reply:
The idea is to keep the process flow more continuous.

Reply:
There's actually a benefit to buffering here: so we can get more efficient batches. So, there's probably some trade-offs. Have you run any performance tests to compare these changes at different maxMem settings to see what effect this has on throughput?